From c28d5a552e266d1605556f347ecae241bd64966c Mon Sep 17 00:00:00 2001
From: Ranbir Singh <Ranbir.Singh3@Dell.com>
Date: Tue, 24 Jan 2023 14:25:43 +0530
Subject: [PATCH 1/1] SecurityPkg/Tcg/Opal/OpalPassword: Fix DEADCODE Coverity
 issue

The PeiDev variable is initialized with the return value from the macro
OPAL_PEI_DEVICE_FROM_THIS which cannot be NULL at that point.

As per Coverity,
Assigning: "PeiDev" = "(OPAL_PEI_DEVICE *)((CHAR8 *)This - 4U)" and
The result of pointer arithmetic "(CHAR8 *)This - 4U" is never null.

Hence, remove the redundant check as it is a DEADCODE.

Cc: Qi Zhang <qi1.zhang@intel.com>
Cc: Rahul Kumar <rahul1.kumar@intel.com>
REF: https://bugzilla.tianocore.org/show_bug.cgi?id=4280
Signed-off-by: Ranbir Singh <Ranbir.Singh3@Dell.com>
---
 SecurityPkg/Tcg/Opal/OpalPassword/OpalPasswordPei.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/SecurityPkg/Tcg/Opal/OpalPassword/OpalPasswordPei.c b/SecurityPkg/Tcg/Opal/OpalPassword/OpalPasswordPei.c
index c998b50be5e8..5bc9270406bf 100644
--- a/SecurityPkg/Tcg/Opal/OpalPassword/OpalPasswordPei.c
+++ b/SecurityPkg/Tcg/Opal/OpalPassword/OpalPasswordPei.c
@@ -98,9 +98,6 @@ SecurityReceiveData (
   OPAL_PEI_DEVICE  *PeiDev;
 
   PeiDev = OPAL_PEI_DEVICE_FROM_THIS (This);
-  if (PeiDev == NULL) {
-    return EFI_DEVICE_ERROR;
-  }
 
   return PeiDev->SscPpi->ReceiveData (
                            PeiDev->SscPpi,
@@ -190,9 +187,6 @@ SecuritySendData (
   OPAL_PEI_DEVICE  *PeiDev;
 
   PeiDev = OPAL_PEI_DEVICE_FROM_THIS (This);
-  if (PeiDev == NULL) {
-    return EFI_DEVICE_ERROR;
-  }
 
   return PeiDev->SscPpi->SendData (
                            PeiDev->SscPpi,
-- 
2.36.1.windows.1

